### PR TITLE
Rudimentary Mech Z-level Travel and Hoverpods

### PR DESCRIPTION
**Pull request by Geevies**
* Added a hoverpod mech.
* Jetpacking downwards while in a mech no longer teleports you through the floor.
* Mechs with hover thrusters can now float over open turfs, as well as move up and down z-levels.

I had a brief discussion with CrimsonShrike, who mentioned the human mob checking its loc, then the mech intercepting it. I did so by making the human check its loc to see if it's a mech, and if it's a mech, transfer the zMove stuff to the mech.
Only issue is that the "X is in the way!" and "Gravity somethingsomething" messages gets sent to the mech instead of the person.

### CODE REVIEW

**Comment by Geevies** [comment]
WIP Until I get a spriter to sprite me hover thrusters.

File: code/modules/multiz/movement.dm [hunk 1]
```diff
@@ -315,6 +319,19 @@
 	if((locate(/obj/structure/disposalpipe/up) in below) || (locate(/obj/machinery/atmospherics/pipe/zpipe/up) in below))
 		return FALSE
 
+/mob/living/heavy_vehicle/can_ztravel(var/direction)
+	if(legs)
+		if(istype(legs, /obj/item/mech_component/propulsion/hover) && legs.motivator.is_functional())
+			if(get_cell().charge < ((legs.power_use * CELLRATE) / 2))
+				return FALSE
+			return TRUE
```
Comment (NonQueueingMatt) [inline code/modules/multiz/movement.dm]: Missing else at the bottom of this?

Comment (NonQueueingMatt) [reply code/modules/multiz/movement.dm]: nvm lmao

**Comment by Geevies** [comment]
!review

**Pull request merged**